lambdas/cognito-user-migration-trigger/index.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const AWS = require('aws-sdk')
+const cognitoIdentityServiceProvider = new AWS.CognitoIdentityServiceProvider({
+  apiVersion: '2016-04-18'
+})
+
+exports.handler = async (event) => {
+  let authReq
+  if (event.triggerSource === 'UserMigration_Authentication') {
+    authReq = cognitoIdentityServiceProvider.adminInitiateAuth({
+      AuthFlow: 'ADMIN_USER_PASSWORD_AUTH',
+      AuthParameters: {
+        USERNAME: event.userName,
+        PASSWORD: event.request.password
+      },
+      ClientId: event.callerContext.clientId,
+      UserPoolId: event.userPoolId
+    }).promise().then(resp => {
+      if (resp.code && resp.message) throw new Error('Bad password')
+      event.response.finalUserStatus = 'CONFIRMED'
+    })
+  } else if (event.triggerSource === 'UserMigration_ForgotPassword') {
+    // Non-error response required to enable password-reset code to be sent to user
+    // `event.response.finalUserStatus` is left alone to trigger reset
+  } else {
```

Comment:
`else if` nothing? What if they need to be migrated and they forgot their password?

Reply:
The new pool will handle the reset. The old pool isn't consulted.

Reply:
Will they still have "all" their old data? (api key, I think is the only thing?) Yes, right?